### PR TITLE
Design Panel: Add badge to layer panel

### LINKED ISSUE #9136: Layers Panel: Add layer count badge to title area
## Description

Sometimes not all layers are visible in the layers panel and it's not super obvious that there are more if you scroll. Add in a badge that displays the count of all the layers in the title area of the layers panel. 

Figma: https://www.figma.com/file/dc9xrVlRhQq4KTE9WQzU1R/_Editor-Sprint-Feature-Level-Designs_?node-id=9954%3A422177 

Design as of Oct 4, 2021:
Screen Shot 2021-10-04 at 10 50 38 AM

## To Do
- [x] - Final UX from design team

## Acceptance Criteria 

- badge displays with layers count, badge stays current even when panel is collapsed. 
- updated tests

### PR DESCRIPTION
**Pull request by samwhale**
## Context

Layer Panel Updates

## Summary

Adds a badge for the layer count on the layer panel

## Relevant Technical Choices

n/a

## To-do

n/a

## User-facing changes

badge

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Open the editor
2. Add things to your page
3. Check out the badge as it changes on the layer panel

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9136

### CODE REVIEW

File: packages/story-editor/src/components/panels/panel/shared/title.js [hunk 1]
```diff
@@ -74,6 +76,25 @@ const Heading = styled.span`
     })};
 `;
 
+const Badge = styled.span(
+  ({ digitLen, theme }) => css`
+    position: relative;
+    height: ${BADGE_DIAMETER}px;
+    width: ${BADGE_DIAMETER + 8 * (digitLen - 1)}px;
```
Comment (samwhale) [inline packages/story-editor/src/components/panels/panel/shared/title.js]: Every time the layer count increases by a digit we add 8px to the width.

The badge's children are absolutely positioned to keep the badge circular. The width will not respect children's length, so as the number increases in width we need to add width to the badge as well.

Lmk if y'all have a better way to make a badge 😄

Comment (swissspidy) [reply packages/story-editor/src/components/panels/panel/shared/title.js]: Can't we just use `NotificationBubble` from the design system...?

File: packages/story-editor/src/components/panels/panel/shared/title.js [hunk 2]
```diff
@@ -229,6 +252,11 @@ function Title({
         >
           {children}
         </Heading>
+        {count !== undefined && (
```
Comment (swissspidy) [inline packages/story-editor/src/components/panels/panel/shared/title.js]: Since you do `count={layers?.length}` it means `count` can also be `null`. And we don't really want to accidentally render `null` here.

I'd do

```suggestion
        {Boolean(count) && (
```

instead

Comment (samwhale) [reply packages/story-editor/src/components/panels/panel/shared/title.js]: `0` is a possibility here too, no? I guess we always have a background layer so maybe we don't have to worry about the `0` case

This is also on the base `panel` so trying to make it agnostic to the layer panel.

Maybe this to account for the `0` case:

```js
(count === 0 || Boolean(count)) && ...
```

File: packages/story-editor/src/components/panels/panel/shared/title.js (hunk 1 above)
Comment (maxyinger) [reply packages/story-editor/src/components/panels/panel/shared/title.js]: I think notification `NotificationBubble` does the same thing too with the fixed width and absolute positioning to make the bg a circle. Should recycle if you can, but you'll probably need to account for the same problem there I think 🤔

**Review by maxyinger — APPROVED** [review]

File: packages/story-editor/src/components/panels/panel/shared/title.js (hunk 1 above)
Comment (samwhale) [reply packages/story-editor/src/components/panels/panel/shared/title.js]: Ah, thanks for the heads up. I didn't realize this existed because it's not in storybook 😭 

image

File: packages/design-system/src/components/notificationBubble/index.js [hunk 3]
```diff
@@ -24,21 +24,40 @@ import PropTypes from 'prop-types';
...
+    background-color: ${variant === 'accent'
+      ? theme.colors.accent.primary
+      : theme.colors.bg[variant]};
     border-radius: ${theme.borders.radius.round};
   `}
   position: relative;
-  height: 24px;
-  width: 24px;
+  height: ${BUBBLE_DIAMETER}px;
+  width: ${({ digitLen }) => BUBBLE_DIAMETER + getBubbleWidth(digitLen)}px;
```
Comment (samwhale) [inline packages/design-system/src/components/notificationBubble/index.js:52]: Expand bubble depending on the number of digits of the bubble

File: packages/design-system/src/components/notificationBubble/index.js [hunk 4]
```diff
@@ -49,20 +68,32 @@ const Inner = styled.span`
     themeHelpers.expandTextPreset(
       ({ paragraph }, sizes) => paragraph[isSmall ? sizes.X_SMALL : sizes.SMALL]
     )({ theme })};
-  color: ${({ theme }) => theme.colors.bg.primary};
+  color: ${({ invertColor, theme }) =>
+    invertColor ? theme.colors.inverted.fg.primary : theme.colors.fg.primary};
```
Comment (samwhale) [inline packages/design-system/src/components/notificationBubble/index.js]: Allow for choosing the text color

File: packages/story-editor/src/components/panels/panel/shared/title.js (hunk 1 above)
Comment (samwhale) [reply packages/story-editor/src/components/panels/panel/shared/title.js]: Should be fixed now!

File: packages/design-system/src/components/notificationBubble/index.js [hunk 5]
```diff
@@ -24,21 +24,40 @@ import PropTypes from 'prop-types';
...
+  return 9 * (numDigits - 1);
+}
+
 const Bubble = styled.div`
-  ${({ theme }) => css`
+  ${({ theme, variant }) => css`
     color: ${theme.colors.fg.primary};
-    background-color: ${theme.colors.accent.primary};
+    background-color: ${variant === 'accent'
+      ? theme.colors.accent.primary
```
Comment (BrittanyIRL) [inline packages/design-system/src/components/notificationBubble/index.js]: Nit: So this is a slippery slope here to introduce a new background color in play that has otherwise only been an active or hover state for interactive bgs. If we're going to start having this violet color be a background i think it should be added to the background colors. Maybe `bg.accent`? 

(also i think this was here before your changes, so apologies for the scope creep)

File: packages/design-system/src/components/notificationBubble/index.js [hunk 6]
```diff
@@ -49,20 +68,32 @@ const Inner = styled.span`
...
 export const NotificationBubble = ({
   notificationCount,
   isSmall,
+  variant = BUBBLE_VARIANTS.ACCENT,
+  invertTextColor,
   ...props
 }) => (
-  <Bubble isSmall={isSmall} {...props}>
-    <Inner isSmall={isSmall}>{notificationCount}</Inner>
+  <Bubble
```
Comment (BrittanyIRL) [inline packages/design-system/src/components/notificationBubble/index.js:83]: Thoughts on using some [transient prop](https://styled-components.com/docs/api#transient-props)s here?

File: packages/design-system/src/components/notificationBubble/stories/index.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+  label {
+    display: flex;
+    align-items: center;
+  }
+`;
+
+export const _default = () => {
+  return (
+    <>
+      <Headline as="h1">{'Notification Bubble'}</Headline>
```
Comment (BrittanyIRL) [inline packages/design-system/src/components/notificationBubble/stories/index.js]: nbd but just so you know this defaults to h1

File: packages/story-editor/src/components/panels/design/layer/karma/layer.karma.js [hunk 8]
```diff
@@ -24,20 +24,54 @@ import { useInsertElement } from '../../../../canvas';
...
+        font: TEXT_ELEMENT_DEFAULT_FONT,
+        content: 'Doo doo doo',
+        x: 40,
+        y: 40,
+        width: 250,
+      })
+    );
+
+    expect(layerPanel.layers.length).toBe(3);
+    expect(layerPanel.panelBadge.textContent).toBe('3');
```
Comment (BrittanyIRL) [inline packages/story-editor/src/components/panels/design/layer/karma/layer.karma.js]: thoughts on checking if the textContent matches `layerPanel.layers.length.toString()`?

**Review by BrittanyIRL — COMMENTED** [review]
Thanks for reusing and refactoring the existing notification bubble, it'll be great to have that aligned going forward. 

It's working well, few nits. When comparing visually Figma's showing the number as weight: 700 font size 14px, i'm seeing 400 weight and 12px when I run this branch locally.

File: packages/design-system/src/components/notificationBubble/stories/index.js (hunk 7 above)
Comment (samwhale) [reply packages/design-system/src/components/notificationBubble/stories/index.js]: I can update the proptype here to not have the `as` prop being required 👍 https://github.com/google/web-stories-wp/blob/main/packages/design-system/src/components/typography/headline/index.js#L49

File: packages/design-system/src/components/notificationBubble/index.js (hunk 5 above)
Comment (samwhale) [reply packages/design-system/src/components/notificationBubble/index.js]: yeah I think that's a good idea 👍

**Review by BrookeGraham — APPROVED** [review]
🗯
